src/openrct2-ui/windows/PatrolArea.cpp

```diff
@@ -190,6 +190,8 @@ class PatrolAreaWindow final : public Window
     void OnToolAbort(rct_widgetindex widgetIndex) override
     {
         hide_gridlines();
+        if (!tool_set(this, 0, Tool::WalkDown))
```

Comment:
This is asking for trouble. If the window is closed then the tool is cancelled and then you recreate the tool on a window that is being destroyed leaving a dangling tool???

Reply:
So, should I remove this hide_gridlines and instead remove the extra show_gridlines that is run when the tool is enabled?